tasks/migrate-removals.ts

```diff
@@ -76,33 +77,34 @@ const summarize = async ({
   inputData: InputData[];
   removalContract: Removal;
 }): Promise<Summary> => {
-  const totalRemovalSupplyOnChain = await outputData.reduce(
-    async (summary, project) => {
-      const updatedSummary = await summary;
-      const projectTotals: BigNumber = await project.tokenIds!.reduce(
-        async (total, removal) => {
-          const updatedTotal = await total;
-          const removalIdSupply = await removalContract.totalSupply(removal);
-          const removalTotal = updatedTotal.add(removalIdSupply);
-          updatedSummary.sum = updatedSummary.sum.add(removalIdSupply);
-          updatedSummary.removals.push(removalTotal);
-          return removalTotal;
-        },
-        Promise.resolve(Zero)
-      );
-      updatedSummary.projects.push(projectTotals);
-      return updatedSummary;
-    },
-    Promise.resolve({
-      sum: Zero,
-      removals: [] as BigNumber[],
-      projects: [] as BigNumber[],
-    })
+  const allRemovalIds = outputData.flatMap((project) => project.tokenIds!);
+  const multicallData = allRemovalIds.map((tokenId) =>
+    removalContract.interface.encodeFunctionData('totalSupply', [tokenId])
+  );
+  const removalAmountsOnChain = await removalContract.callStatic.multicall(
+    multicallData
+  );
+  const bigNumberRemovalAmounts = removalAmountsOnChain.map((amount) =>
+    BigNumber.from(amount)
+  );
+  const totalOnChainSupply = bigNumberRemovalAmounts.reduce(
```

Comment:
This is just fine but you can also avoid needing to write your own reducer with something like `BigNumber.sum.apply(null, bigNumberRemovalAmounts)`